tensilelite/Tensile/TensileCreateLibrary.py

```diff
@@ -69,38 +71,44 @@ def wrapper(*args, **kwargs):
 
     return res
   return wrapper
-################################################################################
-def processKernelSource(kernel, kernelWriterAssembly, ti):
+
+
+class KernelCodeGenResult(NamedTuple):
+    err: int
+    src: str
+    header: Optional[str]
+    name: str
+    targetObjFilename: str
+    isa: IsaVersion
+    wavefrontSize: int
+
+
+def processKernelSource(kernel, kernelWriterAssembly, ti) -> KernelCodeGenResult:
     """
     Generate source for a single kernel.
     Returns (error, source, header, kernelName).
     """
-    try:
-        kernelWriter = kernelWriterAssembly
-        # get kernel name
-        kernelWriter.setTensileInstructions(ti)
-        kernelName = kernelWriter.getKernelFileBase(kernel)
-        (err, src) = kernelWriter.getSourceFileString(kernel)
-        header = kernelWriter.getHeaderFileString(kernel)
-        # will be put in Kernels.h/cpp if None
-        filename = kernel._state.get("codeObjectFile", None)
-
-    except RuntimeError:
-        return (1, "", "", kernelName, None)
+    kernelWriter = kernelWriterAssembly
+    # get kernel name
+    kernelWriter.setTensileInstructions(ti)
+    asmFilename = kernelWriter.getKernelFileBase(kernel)
+    err, src = kernelWriter.getSourceFileString(kernel)
+    header = kernelWriter.getHeaderFileString(kernel)
+    # will be put in Kernels.h/cpp if None
+    objFilename = kernel._state.get("codeObjectFile", None)
+    isa = kernelWriter.isa
+    wavefrontSize = kernelWriter.wavefrontSize
 
-    return (err, src, header, kernelName, filename)
+    return KernelCodeGenResult(err, src, header, asmFilename, objFilename, isa, wavefrontSize)
```

Comment:
Does the wavefrontsize change? Do we have one kernelWriter or many?

Reply:
By the way the code was written, yes, wavefront size will change. Whether it does in practice, I don't know.